### DISCUSSION TITLE
The Mapbox TOS confusion (FUD): Switching to MapLibre

### DISCUSSION THREAD

**Discussion by maxammann** [rnmapbox/maps#1274] [Ideas]
Hello community!

I recently implemented a maps feature and found out that Mapbox switched to a proprietary license with Mapbox 10. Afaik you are bound to there TOS now and have to pay their cloud fees even if you do not use them.

Mapbox is very quiet about their license change and what it means for contributors and users of the library.

It all starts here with a [non-free proprietary, but open-source license. ](https://github.com/mapbox/mapbox-maps-android/blob/main/LICENSE.md). I'm not a lawer, nor very experienced with licenses. Just using my guessing here.

> This license allows developers with a current active Mapbox account to use and modify the Mapbox Maps Android SDK. 

That means you are only allowed to use it if you have an active Mapbox Account. In fact the latest JS library of mapbox no longer works without an active mapbox account. This is also expected for Mapbox Native SDK 10.

You are not allowed to change Billing related code of the new versions. Also you aggree to these terms: https://www.mapbox.com/legal/tos/

A lot of famous people grouped and founded the MapLibre project. Their gl-native fork can be found here: https://github.com/maplibre/maplibre-gl-native

I will soon use react-native-mapbox-gl in an upcoming project. That means we need to have a plan what happens if this repository bumps the Mapbox version to 10. What do the main contributors of this repository think about this? Is this library focused on people who use the Mapbox cloud products?

More information: https://wptavern.com/maplibre-launches-as-official-open-source-successor-to-mapbox-gl-js

**Comment by ferdicus** [comment]
hey @maxammann, would you kindly point to some documentation about 
> Afaik you are bound to there TOS now and have to pay their cloud fees even if you do not use them

Thanks

**Reply by maxammann** [reply]
This is a difficult Story in Fact. Mapbox is very quiet about their license change and what it means for contributors and users of the library.

It all starts here with a [non-free proprietary, but open-source license. ](https://github.com/mapbox/mapbox-maps-android/blob/main/LICENSE.md). I'm not a lawer, nor very experienced with licenses. Just using my guessing here.

> This license allows developers with a current active Mapbox account to use and modify the Mapbox Maps Android SDK. 

That means you are only allowed to use it if you have an active Mapbox Account. In fact the latest JS library of mapbox no longer works without an active mapbox account. This is also expected for Mapbox Native SDK 10.

You are not allowed to change Billing related code of the new versions. Also you aggree to these terms: https://www.mapbox.com/legal/tos/

That means if you have over 25000 users this library is no longer free, even if you do not use maps hosted by mapbox.
(Also you have a problem if you offer your services in Europe as it is basically illegal as you are not allowed to provide US services with personal data of your users.)

**Reply by maxammann** [reply]
Also you require a Mapbox account to download and install mapbox. That means you have to configure an api key before downloading the binaries for android/ios: https://docs.mapbox.com/android/beta/maps/guides/install/

This will be very problematic when upgrading to mapbox sdk 10. Users no longer can just do ǹpm install react-native-mapbox-gl.`. They first have to agree to the Mapbox TOS, configure their api key and then install the native libraries.

**Reply by ferdicus** [reply]
Thanks for the detailed listing of your concerns. 
Unfortunately, I'd have to ask again - because I'm sure, I'm not the only one who want to sift through the whole TOS 😅:  
Where is the paragraph is mentions the `25000` users? 
I wasn't able to find it at a glance.

**Reply by maxammann** [reply]
Oh that is only from the mapbox pricing page: https://www.mapbox.com/pricing/

**Comment by mzu** [comment]
>(Also you have a problem if you offer your services in Europe as it is basically illegal as you are not allowed to provide US services with personal data of your users.)

Is this not covered by [§3.1](https://www.mapbox.com/legal/privacy#[WWStWWSt])? Doesn't sound like this would be an issue for users from the EU.

**Reply by maxammann** [reply]
Probably this is not a real problem, but more a theoretical. Since the EU-US Privacy Shield in no longer in action it is questionable whether you are allowed to share data with an US company as a EU company. But this is not what this Thread should be about.

**Comment by gorbypark** [comment]
I think a discussion about supporting the MapLibre versions of the SDKs is warranted.  At first I wasn't too concerned about the changes to the license and prices, until I noticed (many weeks later...not sure if something has changed since the original announcements) that there is now a Map Seats fee for any app that isnt "open to the public".  The free tier is 3 monthly active users.  That means if you develop an app (such as an internal app for employees, or anything not open to the public for free) you need to pay $4.00 per month per active user.  I have a few apps in production for small to medium buisnesses and non profits, and we are potentially looking at a few thousand to tens of thousands a month in Mapbox fees, which is pretty much a non-starter.  The fees apply even if not using the Mapbox service at all (ie: custom tile servers, etc) like we are.

I wonder if forking this project under the MapLibre banner would be the way to go if there isn't an appetite to support MapLibre?  Any thoughts?

*edit*
It's not still exactly clear as to what exactly a "map seat" is.  It's on the [Mapbox pricing page](https://www.mapbox.com/pricing/) but lacks a clear description as to what it is.

**Reply by maxammann** [reply]
I was also thinking about forking, as so far you are the first person confirming my concern here.

Our company would be willing to put effort into open source development of a library like this.

**Reply by mfazekas** [reply]
@gorbypark it would be great if you could add exact links and quotes to those licensing/pricing issues, otherwise it sounds a bit anecdotal. I did not find much info about the mentioned seats for example, only this: https://docs.mapbox.com/gl-js-seats/

Of course I do understand that the non OSS license issue it's a quite significant issue/risk for esp those who are not using Mapbox services.

**Comment by mfazekas** [comment]
If (or when) MapLibre has [cocoa pods](https://github.com/maplibre/maplibre-gl-native/issues/43) support then we should be able to support it in current master branch. This could be enabled via Cocoapods/gradle variables the same way we support newer non OSS Mapbox releases ([iOS](https://github.com/react-native-mapbox-gl/maps/blob/master/ios/install.md#new-version---since-81rc5), [Andorid](https://github.com/react-native-mapbox-gl/maps/blob/master/android/install.md#mapbox-maps-sdk)) like 6.3.0 at the moment.

I don't think maintainers of this repo has strong opinion about either MapLibre or new Mapbox going forward. It assume it could be either or both depending on community support.

**Reply by gorbypark** [reply]
@mfazekas I'll admit it is very anecdotal and based on reddit comments as they are the only place I can find "map seats" mentioned on the web outside of [Mapbox's pricing page](https://www.mapbox.com/pricing/).  I have reached out to Mapbox's sales team to try and get a better idea of what this all means.  I've edited my comment above to make it clear that no one really knows what a map seat is.

**Reply by maxammann** [reply]
It is so crazy. It seems like mapbox is doing this on purpose. They are causing Fear, uncertainty, and doubt..

**Reply by jaltin** [reply]
> @mfazekas I'll admit it is very anecdotal and based on reddit comments as they are the only place I can find "map seats" mentioned on the web outside of [Mapbox's pricing page](https://www.mapbox.com/pricing/). I have reached out to Mapbox's sales team to try and get a better idea of what this all means. I've edited my comment above to make it clear that no one really knows what a map seat is.

@gorbypark did you get a response from Mapbox's sales team what a seat means?

**Comment by mfazekas** [comment]
FWIW PR #1311 allows one to use RNMBGL with MapLibre on iOS

**Comment by gorbypark** [comment]
Unfortunately they have not replied (yet? It’s been two weeks so I’m not
holding my breath ).  I’ve tried their online chat system as well as
emailing their sales team.

**Reply by jaltin** [reply]
Thanks for the quick reply. Really a pity that it's so hard to find info from them.

**Comment by ferdicus** [comment]
Curious: has anyone already used MapLibre?

**Reply by maxammann** [reply]
Not yet but a few people are already using it directly. There is already a for flutter: https://github.com/m0nac0/flutter-maplibre-gl
